### PR TITLE
Homework 5

### PR DESCRIPTION
**Pull request by OlgaPoddubnaia**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -141,6 +158,45 @@
...
+            <version>3.141.59</version>
+            <scope>compile</scope>
+        </dependency>
+
+        <dependency>
+            <groupId>org.seleniumhq.selenium</groupId>
+            <artifactId>selenium-opera-driver</artifactId>
+            <version>3.141.59</version>
+            <scope>compile</scope>
+        </dependency>
```
Comment (DmitryKhodakovsky) [inline pom.xml:199]: Why do you need these drivers as separate dependencies?

File: src/test/java/hw5/steps/ThenStepDef.java [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+        softAssert.assertEquals(homePagePO.shouldHaveTitle(), "Home Page");
+        softAssert.assertAll();
+    }
+
+    @Then("Logged user name equals \"Roman Iovlev\"")
+    public void checkIsUserLogged() {
+        SoftAssert softAssert = new SoftAssert();
+        softAssert.assertEquals(loginPage.getUserNameAfterLogin(),
+                ConfProperties.getProperty("LOGGED_USER_NAME"));
+        softAssert.assertAll();
```
Comment (DmitryKhodakovsky) [inline src/test/java/hw5/steps/ThenStepDef.java:32]: What is the reason of using softAssertions here?

File: src/test/java/hw5/steps/ThenStepDef.java [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+    @Then("\"User Table\" page should be opened")
+    public void checkUserTableUrl() {
+        SoftAssert softAssert = new SoftAssert();
+        softAssert.assertEquals(userTablePO.getUserTablePageUrl(), ConfProperties
+                .getProperty("USER_TABLE_URL"));
+        softAssert.assertAll();
+    }
+
+
+    @Then("6 Number Type Dropdowns should be displayed on Users Table on User Table Page")
```
Comment (DmitryKhodakovsky) [inline src/test/java/hw5/steps/ThenStepDef.java:102]: Why is number of dropdown hardcoded?

File: src/test/java/hw5/steps/ThenStepDef.java [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+    public void areSixNumberTypeDropdownsDisplayed() {
+        SoftAssert softAssert = new SoftAssert();
+        softAssert.assertTrue(userTablePO.areThereSixNumberTypeDropdowns());
+        for (int i = 0; i < userTablePO.numberTypeDropdownsSize(); i++) {
+            softAssert.assertTrue(userTablePO.areNumberTypeDropdownsDisplayed(i));
+        }
+        softAssert.assertAll();
+    }
+
+    @Then("6 Usernames should be displayed on Users Table on User Table Page")
```
Comment (DmitryKhodakovsky) [inline src/test/java/hw5/steps/ThenStepDef.java:112]: Why is number of the users hardcoded?

File: src/test/java/hw5/steps/ThenStepDef.java [hunk 5]
```diff
@@ -0,0 +1,169 @@
...
+    public void areSixCheckboxesDisplayed() {
+        SoftAssert softAssert = new SoftAssert();
+        softAssert.assertTrue(userTablePO.areThereSixCheckboxes());
+        for (int i = 0; i < userTablePO.checkboxesSize(); i++) {
+            softAssert.assertTrue(userTablePO.areCheckboxesDisplayed(i));
+        }
+        softAssert.assertAll();
+    }
+
+    @Then("1 log row has \"Vip: condition changed to true\" text in log section")
```
Comment (DmitryKhodakovsky) [inline src/test/java/hw5/steps/ThenStepDef.java:142]: Why is expected text of the log row is hardcoded?

File: src/test/java/hw5/steps/WhenStepDef.java [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+        headerMenuOfHomePageAfterLoginPO.clickServiceInDropdown();
+    }
+
+    @When("I open Different Elements page through the header menu->Service")
+    public void openDifferentElementsPage() {
+        headerMenuOfHomePageAfterLoginPO.openDifferentElementsPage();
+    }
+
+
+    @When("I select water, wind checkboxes")
```
Comment (DmitryKhodakovsky) [inline src/test/java/hw5/steps/WhenStepDef.java:18]: What will you actions if you need select more checkboxes?

File: src/test/resources/hw5/Cucumber_Exercise_2.feature [hunk 7]
```diff
@@ -0,0 +1,26 @@
+Feature: Exercise 2 Homework5
+
+  Scenario:User Table Page test
+    Given I open JDI GitHub site
+    And I login as user "Roman Iovlev"
+    When I click on "Service" button in Header
+    And I click on "User Table" button in Service dropdown
```
Comment (DmitryKhodakovsky) [inline src/test/resources/hw5/Cucumber_Exercise_2.feature:7]: `Service` and `User Table` should be as parameters

File: src/test/resources/hw5/Cucumber_Exercise_2.feature [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+    Then "User Table" page should be opened
+    And 6 Number Type Dropdowns should be displayed on Users Table on User Table Page
+    And 6 Usernames should be displayed on Users Table on User Table Page
+    And 6 Description texts under images should be displayed on Users Table on User Table Page
+    And 6 checkboxes should be displayed on Users Table on User Table Page
```
Comment (DmitryKhodakovsky) [inline src/test/resources/hw5/Cucumber_Exercise_2.feature:12]: The numbers of the elements should be as parameters

File: src/test/resources/hw5/Cucumber_Exercise_3.feature [hunk 9]
```diff
@@ -0,0 +1,9 @@
+Feature: Exercise 3 Homework5
+
+  Scenario: log row for VIP checkbox test
+    Given I open JDI GitHub site
+    And I login as user "Roman Iovlev"
+    And I click on "Service" button in Header
+    And I click on "User Table" button in Service dropdown
+    When I select 'vip' checkbox for "Sergey Ivan"
+    Then 1 log row has "Vip: condition changed to true" text in log section
```
Comment (DmitryKhodakovsky) [inline src/test/resources/hw5/Cucumber_Exercise_3.feature:9]: Number of log rows and text should be as parameter

File: src/test/resources/hw5/Cucumber_Exercise_1.feature [hunk 10]
```diff
@@ -0,0 +1,19 @@
+Feature: Exercise 1 Homework5
+
+  Scenario: Different elements page test
+    Given I open JDI GitHub site
+    Then The site has proper Url
```
Comment (DmitryKhodakovsky) [inline src/test/resources/hw5/Cucumber_Exercise_1.feature:5]: What is the `Proper URL` for the site?

File: src/test/resources/hw5/Cucumber_Exercise_1.feature [hunk 11]
```diff
@@ -0,0 +1,19 @@
...
+    Then The site has proper title
```
Comment (DmitryKhodakovsky) [inline src/test/resources/hw5/Cucumber_Exercise_1.feature:6]: What is the `Proper Title` for the site?

**Review by DmitryKhodakovsky — CHANGES_REQUESTED** [review]

File: pom.xml (hunk 1 above)
Comment (OlgaPoddubnaia) [reply pom.xml:199]: I tried to delete them for a several times but then code stops to work(all selenium annotations etc) and IntelliJ requests to load dependency so this happens.

Comment (OlgaPoddubnaia) [reply pom.xml:199]: Also I think you talk about other drivers : selenium-api, selenium-support

File: src/test/java/hw5/steps/ThenStepDef.java (hunk 2 above)
Comment (OlgaPoddubnaia) [reply src/test/java/hw5/steps/ThenStepDef.java:32]: I misunderstood the example from the lecture. there is no need to use most  of softAsserts except with loops or other multiple softAsserts at one method.

File: src/test/resources/hw5/Cucumber_Exercise_2.feature (hunk 7 above)
Comment (OlgaPoddubnaia) [reply src/test/resources/hw5/Cucumber_Exercise_2.feature:7]: Made Service and User Table as parameters

File: src/test/resources/hw5/Cucumber_Exercise_2.feature (hunk 8 above)
Comment (OlgaPoddubnaia) [reply src/test/resources/hw5/Cucumber_Exercise_2.feature:12]: changed numbers to parameters (4 times)

File: src/test/resources/hw5/Cucumber_Exercise_3.feature (hunk 9 above)
Comment (OlgaPoddubnaia) [reply src/test/resources/hw5/Cucumber_Exercise_3.feature:9]: changed number of log rows and text to parameters

File: src/test/java/hw5/steps/ThenStepDef.java (hunk 3 above)
Comment (OlgaPoddubnaia) [reply src/test/java/hw5/steps/ThenStepDef.java:102]: changed number to parameter

File: src/test/java/hw5/steps/ThenStepDef.java (hunk 4 above)
Comment (OlgaPoddubnaia) [reply src/test/java/hw5/steps/ThenStepDef.java:112]: changed number to parameter

File: src/test/java/hw5/steps/ThenStepDef.java (hunk 5 above)
Comment (OlgaPoddubnaia) [reply src/test/java/hw5/steps/ThenStepDef.java:142]: changed text to parameter

File: src/test/resources/hw5/Cucumber_Exercise_1.feature (hunk 11 above)
Comment (OlgaPoddubnaia) [reply src/test/resources/hw5/Cucumber_Exercise_1.feature:6]: made title as a parameter so the step now is "Then The site has title 'Home Page'"

File: src/test/resources/hw5/Cucumber_Exercise_1.feature (hunk 10 above)
Comment (OlgaPoddubnaia) [reply src/test/resources/hw5/Cucumber_Exercise_1.feature:5]: changed this step by putting url in it

File: src/test/java/hw5/steps/WhenStepDef.java (hunk 6 above)
Comment (OlgaPoddubnaia) [reply src/test/java/hw5/steps/WhenStepDef.java:18]: changed water, wind  to parameters also changed method on page object

**Review by DmitryKhodakovsky — APPROVED** [review]
Работа принята

**Pull request merged**